src/Atmos/Model/AtmosModel.jl

```diff
@@ -121,14 +123,19 @@ Where
 """
 @inline function flux_nondiffusive!(m::AtmosModel, flux::Grad, state::Vars,
                                     aux::Vars, t::Real)
-  ρinv = 1/state.ρ
+  ρ = state.ρ
+  ρinv = 1/ρ
   ρu = state.ρu
   u = ρinv * ρu
+  z  = altitude(m.orientation, aux)
 
   # advective terms
-  flux.ρ   = ρu
-  flux.ρu  = ρu .* u'
-  flux.ρe  = u * state.ρe
+  usub = subsidence_speed(m.subsidence, z)
+  vertical_unit_vector = ∇gravitational_potential(m.orientation, aux)
```

Comment:
I think it's better when `vertical_unit_vector` is provided by `orientation.jl`. You may compute it from the gravitational potential there (but need to divide by `g`!).

Reply:
>I think it's better when vertical_unit_vector is provided by orientation.jl. 

`∇gravitational_potential` is a function in `orientation.jl`, but I'll add a new method specifically for the unit vector.

> You may compute it from the gravitational potential there (but need to divide by g!)

Good catch!